Docs/source/install/hpc/narval.rst

```diff
@@ -0,0 +1,153 @@
+.. _building-narval:
+
+Narval (Digital Research Alliance of Canada)
+========================================================
+
+The `Narval cluster <https://www.calculquebec.ca/en/nouvelle/narval-a-new-supercomputer-dedicated-to-scientific-research/>`__ is located at McGill University.
+
+Introduction
+------------
+
+If you are new to this system, **please see the following resources**:
+
+* `Narval Wiki <https://docs.alliancecan.ca/wiki/Narval/en>`__
+* `Infrastructures and Services <https://www.calculquebec.ca/en/academic-research-services/infrastructures-services/#ouvert>`__
+
+Through Narval we have access to CPU and GPU nodes.
+
+Installation
+------------
+
+Narval has no WarpX preinstalled, we need to compile it from scratch. Since ``$HOME`` directory is designed for compiling codes, the following operations will be executed in ``$HOME`` directory.
+
+Download WarpX
+^^^^^^^^^^^^^^
+
+.. code-block:: bash
+
+    cd $HOME
+    git clone https://github.com/ECP-WarpX/WarpX.git
+
+Building WarpX
+^^^^^^^^^^^^^^
+
+* To build it with CPU backend, load the following module
+
+.. code-block:: bash
+
+    module load StdEnv/2023
+
+This environment contains ``gcc/12.3, openmpi/4.1.5`` and other usefule modules. Now build WarpX as described in `documentation <https://warpx.readthedocs.io/en/latest/install/cmake.html>`__
+
+.. code-block:: bash
+
+    cmake -S $HOME/WarpX -B $HOME/WarpX/build -DWarpX_DIMS="1;2;3;RZ" \
+        -DWarpX_COMPUTE=OMP \
+        -DWARX_MPI=ON \
+    cmake --build build -j 4
+
+
+* To build it with GPU backend, load the following modules
+
+.. code-block:: bash
+
+    module load StdEnv/2023 cuda/12.2
+
+The only thing we need to change is the ``-DWarpX_COMPUTE`` flag,
+
+.. code-block:: bash
+
+    cmake -S $HOME/WarpX -B $HOME/WarpX/build -DWarpX_DIMS="1;2;3;RZ" \
+        -DWarpX_COMPUTE=CUDA \
+        -DWarpX_MPI=ON \
+    cmake --build build -j 4
+
+Python Bindings
+^^^^^^^^^^^^^^^
+
+1. Create virtualenv
+
+.. code-block:: bash
+
+    virtualenv --system-site-packages $HOME/.venvs/warpx
+    source $HOME/.venvs/warpx/bin/activate
+
+2. Compile WarpX with extra flag ``-DWarpX_PYTHON=ON``, and install ``pywarpx`` into the virtual environment
```

Comment:
Can you add a `install_dependencies.sh` script as we do on other systems?

That way, we can add more dependencies for things missing in the module system (e.g., better IO and vis) and keep the docs consistent and automated between systems.